dynamiqs/time_array.py

```diff
@@ -30,6 +31,9 @@ def constant(array: ArrayLike) -> ConstantTimeArray:
         _(time-array object of shape (..., n, n) when called)_ Callable object
             returning $O_0$ for any time $t$.
     """
+    if isinstance(array, SparseQArray):
+        return ConstantTimeArray(array)
+
```

Comment:
Here (and below), we should still keep the call to `check_shape` even if the input is a `SparseQArray`. So only the `jnp.asarray` should be discarded here.